### PR TITLE
Manage User Property

### PR DESCRIPTION
**Pull request by anim-13**
In this PR, I had implemented following features:
1. User can create property
2. User can edit property
3. User can delete property.

Trello Link: https://trello.com/c/Hz3pASvc

Screenshot from 2021-03-19 14-55-00
Screenshot from 2021-03-19 14-55-22

### CODE REVIEW

File: app/controllers/estates_controller.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      flash.now.alert = "Not Created"
+      render :new
+    end
+  end
+  
+  def edit; end
+
+  def update    
+    if @estate.update(estate_params)
+      redirect_to user_estates_path
```
Comment (kalpana19sharma) [inline app/controllers/estates_controller.rb:29]: Set proper flash messages.

File: app/controllers/estates_controller.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  def estate_params
+    params.require(:estate).permit(:name, :address, :area, :bedroom, :hall , :kitchen, :park)
+  end
+
+  def set_user
+    @user = current_user
+  end
+
+  def set_estate
+    @estate = Estate.find_by_id(params[:id])
```
Comment (kalpana19sharma) [inline app/controllers/estates_controller.rb]: Use associations

File: app/models/estate.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Estate < ApplicationRecord
+  belongs_to :user
+
+  validates :name, presence: true
+  validates :address, presence: true
+  validates :area, presence: true
+  validates :bedroom, presence: true	
+  validates :hall, presence: true
+  validates :kitchen, presence: true	
+  validates :park, presence: true
```
Comment (kalpana19sharma) [inline app/models/estate.rb]: Can we combine these validations.

File: app/views/estates/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+          <th scope = "row"><h6><%= index_no %>.</h6> </td> 
+          <td><h6><%= est.name %></h6></td>
+          <td><h6><%= est.address %></h6></td>
+          <td><h6><%= est.area %></h6></td>
+          <td><h6><%= est.bedroom %></h6></td> 
+          <td><h6><%= est.hall %></h6></td>
+          <td><h6><%= est.kitchen %></h6></td>
+          <td><h6><%= est.park %></h6></td>
+          <td>
+            <%= link_to "Edit", edit_user_estate_path(:id => est.id) %> | <%= link_to "Delete", user_estate_path(:id => est.id),
```
Comment (kalpana19sharma) [inline app/views/estates/index.html.erb]: Don't use `=>` syntax

File: app/views/estates/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    <%= f.number_field :area, class: "text_field", placeholder: "Area" %>
+  </div>
+  <div class = "labels text-body" >
+    <%= f.label :Bedroom, class: "text-white" %><br>
+    <%= f.number_field :bedroom, class: "text_field", placeholder: "Bedroom" %>
+  </div>
+  <div class = "labels text-body" >
+    <%= f.label :Hall, class: "text-white" %><br>
+    <%= f.number_field :hall, class: "text_field", placeholder: "Hall" %>
+  </div>
```
Comment (kalpana19sharma) [inline app/views/estates/new.html.erb]: Is it possible to extract form fields in a partial and render that partial in new and edit views?

**Review by kalpana19sharma — CHANGES_REQUESTED** [review]

File: app/models/estate.rb (hunk 3 above)
Comment (anim-13) [reply app/models/estate.rb]: yes

File: app/views/estates/new.html.erb (hunk 5 above)
Comment (anim-13) [reply app/views/estates/new.html.erb]: Done

File: app/controllers/estates_controller.rb [hunk 6]
```diff
@@ -0,0 +1,53 @@
+class EstatesController < ApplicationController
+  before_action :set_user, only: [:new, :create, :edit, :update, :destroy]
+  before_action :set_estate, only: [:edit, :update, :destroy]
```
Comment (kalpana19sharma) [inline app/controllers/estates_controller.rb]: Use `%i` syntax

File: app/controllers/estates_controller.rb [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    end
+  end
+
+  def destroy
+    @estate.destroy
+    redirect_to user_estates_path
+  end
+
+  private
+  def estate_params
```
Comment (kalpana19sharma) [inline app/controllers/estates_controller.rb:43]: Add an empty line above method definition

File: app/views/estates/_estate_form.html.erb [hunk 8]
```diff
@@ -0,0 +1,33 @@
+<%= form_for ([ @user, @estate]) do |f| %>
```
Comment (kalpana19sharma) [inline app/views/estates/_estate_form.html.erb]:
```suggestion
<%= form_for ([@user, @estate]) do |f| %>
```

**Review by kalpana19sharma — CHANGES_REQUESTED** [review]

**Review by kalpana19sharma — APPROVED** [review]

**Pull request merged**